util/aws_retryer.go

```diff
@@ -0,0 +1,135 @@
+package util
+
+import (
+	"math/rand"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+type ConvoyAWSRetryer struct {
+	NumMaxRetries    int
+	MinDelay         int
+	MinThrottleDelay int
+	MaxDelay         int
+}
+
+// DefaultConvoyAWSRetryer creates a new ConvoyAWSRetryer with sensible defaults
+func DefaultConvoyAWSRetryer() *ConvoyAWSRetryer {
+	return &ConvoyAWSRetryer{NumMaxRetries: 10, MinDelay: 200, MinThrottleDelay: 500, MaxDelay: 30000}
+}
+
+// MaxRetries returns the number of maximum returns the service will use to make
+// an individual API request.
+func (c ConvoyAWSRetryer) MaxRetries() int {
+	return c.NumMaxRetries
```

Comment:
rename to MaxRetries

Reply:
Cant :( theres a method called MaxRetries(), i wanted to change it too.